internal/coreinternal/processor/filtermetric/filtermetric.go

```diff
@@ -15,53 +15,168 @@
 package filtermetric // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filtermetric"
 
 import (
+	"fmt"
+
 	"go.opentelemetry.io/collector/model/pdata"
+	conventions "go.opentelemetry.io/collector/model/semconv/v1.6.1"
 	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterconfig"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filtermatcher"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterset"
+)
+
+type FilterType int
+
+const (
+	INCLUDE FilterType = iota + 1
+	EXCLUDE
 )
 
 type Matcher interface {
 	MatchMetric(metric pdata.Metric) (bool, error)
 }
+type AttrMatcher interface {
```

Comment:
Could we instead unify Matcher and AttrMatcher (possibly into a Matcher to keep it consistent)?

Reply:
I have another branch that entirely removes the custom filtermetric config and merges it with the filterconfig one but I was gonna leave that for a follow-up pr...

Reply:
Awesome @hughsimpson! If it would make PR considerably larger (say >600 lines changed in total), we might prefer to keep it separate. I see the value in doing it at once though as we would keep the interfaces consistent

Reply:
Diff is `17 files changed, 377 insertions(+), 292 deletions(-)` ATM so it's on the border of that... but go hard or go home I suppose

Reply:
I think it's worth doing it as one PR (also, I guess we would see some extra lines in filterprocessor's README.md)